lib/lookup/src/path.lalrpop

```diff
@@ -0,0 +1,33 @@
+use crate::{Lookup, Segment, Field};
+use std::str::FromStr;
+
+grammar;
+
+pub Lookup: Lookup<'input> = {
+  <s: PathSegment> <l: Lookup?> => match l {
+    None => Lookup::from(s),
+    Some(mut l) => {
+      l.push_front(s);
+      l
+    }
+  }
+}
+
+PathSegment: Segment<'input> = {
+  "."? <Field> => Segment::field(<>),
+  "[" <Index> "]" => Segment::index(<>),
+  "."? "(" <v:(<Field> "|")+> <e:Field> ")" => {
+        let mut v = v;
+        v.push(e);
+        Segment::coalesce(v)
+  }
+}
+
+Field: Field<'input> = {
+   <s:r"[a-zA-Z_][0-9a-zA-Z_]*"> => Field::from(s),
```

Comment:
In VRL, it is acceptable to have a field `0foo` (this was recently changed), which this regexp doesn't allow for.

(note that a field `0` in VRL still has to be quoted: `.0foo."0"`)